lib/infomon.rb

```diff
@@ -0,0 +1,515 @@
+            done = true
+          end
+          nil
+        elsif server_string == "<pushBold/>\r\n"
+          hide_lines = true
+          nil
+        else
+          server_string
+        end
+      }
+      DownstreamHook.add('infomon_society', action)
+      echo 'checking society...'
+      put 'society'
+      wait_until { done }
+
+      done = false
+      action = proc { |server_string|
+        if server_string =~ /You currently have .*? citizenship in|You don't seem to have citizenship\./
+          DownstreamHook.remove('infomon_citizenship')
+          done = true
+          nil
+        else
+          server_string
+        end
+      }
+      DownstreamHook.add('infomon_citizenship', action)
+      echo 'checking citizenship...'
+      put 'citizenship'
+      wait_until { done }
+
+      silence_me
+
+    end
+
+    if $infomon_values['active_spells'].empty?
+      echo 'checking active spells...'
+      $process_legacy_spell_durations = true
+    end
+    echo 'done'
+
+    #
+    # Load spell timers
+    #
+    $infomon_values['active_spells'].each_pair do |spell_num, timeleft|
+      if (spell = Spell[spell_num.to_i])
+        if defined?(spell.real_time) && spell.real_time
+          timeleft = (timeleft - Time.now.to_f) / 60.0
+          if timeleft.positive?
+            spell.timeleft = timeleft
+            spell.active = true
+          end
+        else
+          spell.timeleft = timeleft
+          spell.active = true
+        end
+      else
+        echo "spell not loaded: #{spell_num}"
+      end
+    end
+    Spellsong.load_serialized = $infomon_values['Spellsong'] if $infomon_values['Spellsong']
+
+    #
+    # Register ;magic and ;banks commands moved
+    #
+
+    $infomon_sleeping = false
+    $infomon_bound = false
+    $infomon_silenced = false
+    $infomon_calmed = true
+    $infomon_cutthroat = false
+    #
+    # Save function
+    #
+    def self.save_proc
+      @save_value = $infomon_values
+      respond @save_value if $infomon_debug
+      @save_value['active_spells'] = {}
+      Spell.active.each do |spell|
+        @save_value['active_spells'][spell.num.to_s] = if defined?(spell.real_time) && spell.real_time
+                                                         Time.now.to_f + (spell.timeleft * 60)
+                                                       else
+                                                         spell.timeleft
+                                                       end
+      end
+      @save_value['Spellsong'] = Spellsong.serialize
+      DB_Store.save('infomon', @save_value)
+    end
+
+    #
+    # Save current status every five minutes in case of crash
+    #
+    Thread.new do
+      loop do
+        sleep 300
+        Infomon::Monitor.save_proc
+      end
+    rescue StandardError
+      echo $ERROR_INFO
+      echo $ERROR_INFO.backtrace[0..1]
+    end
+    #
+    # Save current status on exit
+    #
+    before_dying do
```

Comment:
how does this even work at the top-level?

Reply:
Removed this section completely in latest commits.